apps/site/tests/e2e/general-behavior.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { importLocale } from '@node-core/website-i18n';
+import { test, expect, type Page } from '@playwright/test';
+
+// TODO(@avivkeller): It would be ideal for all the Test IDs to not exist in the
+//                    ui-components package, and instead be passed as props.
+const testIds = {
+  themeToggle: 'theme-toggle',
+  languageDropdown: 'language-selector',
+  languageOptions: 'language-options',
+  navLinks: 'nav-links',
+  mobileMenuToggle: 'mobile-menu-toggle',
+};
+
+// These are inherited from Orama, so they don't have test IDs. Instead, we use the element names directly
+const selectors = {
+  searchButton: 'orama-button',
+  searchInput: 'orama-input',
+  searchResults: 'orama-search-results',
+};
+
+// Helper functions
+const getTheme = (page: Page) =>
+  page.evaluate(() => document.documentElement.dataset.theme);
+
+const openLanguageMenu = async (page: Page) => {
+  await page.getByTestId(testIds.languageDropdown).first().click();
```

Comment:
per my other comment, I think it'd be [getByRole](https://playwright.dev/docs/locators#locate-by-role)

```suggestion
  await page.getByRole('button', { name: 'Choose Language' }).click();
```

I think this has the benefit of ensuring only one, if not, it errors.

this is a nonblocking nit but i think something we could apply throughout to be more accessible-minded